src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/utils/SessionTemplateComparator.kt

```diff
@@ -120,42 +103,27 @@ class SessionTemplateComparator(private val sessionDatesUtil: SessionDatesUtil)
   }
 
   private fun hasCommonSessionLocations(newSessionDetails: SessionDetailsDto, existingSessionDetails: SessionDetailsDto): Boolean {
-    return if (newSessionDetails.permittedLocationGroups.isEmpty() || existingSessionDetails.permittedLocationGroups.isEmpty()) {
-      true
-    } else if (newSessionDetails.permittedLocationGroups.isNotEmpty() && existingSessionDetails.permittedLocationGroups.isNotEmpty()) {
-      val existingSessionLocations = getPermittedSessionLocationDtos(existingSessionDetails)
-      val newSessionLocations = getPermittedSessionLocationDtos(newSessionDetails)
-
-      hasLocationMatch(newSessionLocations, existingSessionLocations)
-    } else {
-      false
-    }
+    val existingSessionLocations = sessionTemplateUtil.getPermittedSessionLocations(existingSessionDetails.permittedLocationGroups)
+    val newSessionLocations = sessionTemplateUtil.getPermittedSessionLocations(newSessionDetails.permittedLocationGroups)
+
+    return sessionLocationMatcher.hasAnyMatch(newSessionLocations, existingSessionLocations) ||
+      sessionLocationMatcher.hasAnyMatch(existingSessionLocations, newSessionLocations)
```

Comment:
be careful changing code out side of ticket scope,...

Reply:
mandated by the code changes to this PR